### PR TITLE
hopefully works

### LINKED ISSUE #260: Store key using Social Sharing
Add a button to save key with Social Sharing.

The button should open multi-select of users to whom the key shares would be sent.
The shares should be send in private message as a file. Name of a file should be  `"#{Key_owner_name}-#{Key_owner_short_code}.social_part"`

User should create at least 5 shares to make the key recoverable.
This could be done in several takes.

### PR DESCRIPTION
**Pull request by illia-kuznietsov**
closes #260

### CODE REVIEW

File: lib/chat_web/live/main_live/index.html.heex [hunk 1]
```diff
@@ -297,6 +303,20 @@
...
+        </p>
+        <.live_component module={ChatWeb.MainLive.Layout.SecretSharingForm} id="secret" me={@me} />
+
+        <button
+          phx-click={hide_modal("secret-share-popup")}
+          class="w-full mr-1 h-12 border rounded-lg border-black/10"
+        >
+          Cancel
+        </button>
+      </.modal>
```
Comment (sviatosluv) [inline lib/chat_web/live/main_live/index.html.heex:319]: You can also represent the entire modal as a LiveComponent. See `open_modal/3`. 
This is not necessary, but just FYI

File: lib/chat_web/live/main_live/layout/secret_sharing_form.ex [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      |> Enum.to_list()
+      |> Enum.sort_by(&elem(&1, 1))
+
+    {:ok,
+     socket
+     |> assign(:secret_holders, [])
+     |> assign(:user_options, users)
+     |> assign(:selected_pb_key, nil)
+     |> assign(:selected_kind, nil)
+     |> assign(:threshold, 3)}
```
Comment (sviatosluv) [inline lib/chat_web/live/main_live/layout/secret_sharing_form.ex:29]: You may use `ok/1` and `noreply/1` helpers in the callback responses.
This is not necessary, but just FYI

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ Here we write upgrading notes for brands. It's a team effort to make them as
 straightforward as possible.
 
 ### Added
+- Secret Sharing integrated into client via new form in logout
+- After clicking the button, secret shares will be sent to chosen users via text message
```
Comment (sergey-lukianov) [inline CHANGELOG.md:15]: one line per PR, please

File: lib/chat_web/live/main_live/layout/secret_sharing_form.ex [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+  @impl Phoenix.LiveComponent
+  def mount(socket) do
+    users =
+      Registry.all()
+      |> Stream.map(fn {pub_key, %Card{} = user} -> {pub_key, user.name} end)
+      |> Enum.to_list()
+      |> Enum.sort_by(&elem(&1, 1))
+
+    {:ok,
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/layout/secret_sharing_form.ex:23]: use `ok/1` helper

File: lib/chat_web/live/main_live/layout/secret_sharing_form.ex [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  alias Chat.Messages.Text
+  alias Chat.User.Registry
+
+  @impl Phoenix.LiveComponent
+  def mount(socket) do
+    users =
+      Registry.all()
+      |> Stream.map(fn {pub_key, %Card{} = user} -> {pub_key, user.name} end)
+      |> Enum.to_list()
+      |> Enum.sort_by(&elem(&1, 1))
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/layout/secret_sharing_form.ex:21]: 1. This list includes myself, which is wrong
2. `UsersBrocker` is a better source of users

File: lib/chat_web/live/main_live/layout/secret_sharing_form.ex [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+     |> assign(:threshold, 3)}
+  end
+
+  @impl Phoenix.LiveComponent
+  def handle_event("select_user", %{"pub-key" => pub_key, "kind" => kind}, socket)
+      when kind in ["share-holders", "rest"] do
+    {:noreply,
+     socket
+     |> assign(:selected_pb_key, decode_pub_key(pub_key))
+     |> assign(:selected_kind, kind)}
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/layout/secret_sharing_form.ex:38]: use `noreply` helper please

File: lib/chat_web/live/main_live/layout/secret_sharing_form.ex [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    secret_sharer = socket.assigns.me
+
+    shares =
+      secret_sharer.private_key
+      |> Enigma.hide_secret_in_shares(length(share_holders), socket.assigns.threshold)
+
+    for {{key, _name}, share} <- Enum.zip(share_holders, shares) do
+      dialog = Dialog.start(secret_sharer, %Chat.Card{pub_key: key})
+
+      Text.new(share |> Base.encode64(), DateTime.utc_now() |> DateTime.to_unix())
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/layout/secret_sharing_form.ex:82]: bad time source. See user monotonic_offset

**Review by sergey-lukianov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**